src/index.ts

```diff
@@ -80,11 +83,27 @@ const loadFile = (filePath: string): Record<string, unknown> => {
   return {};
 };
 
-const convertString = (value: string): string | number | boolean => {
-  if (value.toLowerCase() === 'true') return true;
-  if (value.toLowerCase() === 'false') return false;
-  if (/^\d+\.\d+$/.test(value)) return Number.parseFloat(value);
-  if (/^\d+$/.test(value)) return Number.parseInt(value, 10);
+const unwrapValue = (value: ConfigValue): ConfigValue => {
```

Comment:
This function now recursively produces a `ConfigValue` in the same structure as the input, basically just with string values parsed to booleans | numbers.

Reply:
My only suggestion would be maybe adding a `maxDepth` or something to prevent a bad config value from causing an infinite loop. Not sure how realistic that is though.

Reply:
Just tried it and it seems this already handles that nicely due to how circular references are handled.

1. It's not possible to have a circular reference encoded in JSON.  There are some specs out there but they require custom parsers for that.
2. Creating a circular reference in the JS configs is possible, but since these values are never serialized, they're handled perfectly fine:

config:
```ts
const config: Record<string, unknown> = {
  VALUE: 1,
};

config['OH_NO'] = config;

export default config;
```

Loading this config is fine, and printing `OH_NO` returns an object reference!

```js
console.log(config.OH_NO);
```

<img width="388" alt="image" src="https://github.com/user-attachments/assets/02706786-8115-4555-92f5-c84ca2125e1a">

The same thing happens with `config.OH_NO.OH_NO.OH_NO.OH_NO`, etc.

I can't think of any other ways we could get circular references.  I can definitely add a max depth though if we're concerned about that potential happening something though.

Reply:
> `config.OH_NO.OH_NO.OH_NO.OH_NO`

had a good chuckle.